src/TraceEvent/Stacks/RecursionGuard.cs

```diff
@@ -2,31 +2,45 @@
 
 namespace Microsoft.Diagnostics.Tracing.Stacks
 {
-    /// <summary>
-    /// This structure provides a clean API for a lightweight recursion stack guard to prevent StackOverflow exceptions
-    /// We do ultimately do a stack-overflow to prevent infinite recursion, but it is now under our
-    /// control and much larger than you may get on any one thread stack.  
-    /// </summary>
-    internal struct RecursionGuard
+    public static class RecursionGuardConfiguration
     {
+        private static ushort _singleThreadRecursionLimit = 400;
+        private static ushort _maxResets = 80;

```

Comment:
If we use configuration, only `_maxResets` needs to be configurable. This value is only used on the "slow" path, where `_singleThreadRecursionLimit` is used on the "fast" path.

Reply:
You're right that only `_maxResets` needs to be configured.  We've not hit an issue with the single thread recursion limit.  I'll make that change.

I spent a couple of days thinking about your proposal.  In the end, I still very much want a "backstop" to prevent infinite recursion, even in release builds where TraceEvent gets used in automated data pipelines.  I would much rather folks hit an exception and address it than end up with infinite recursion and have to detect it.  With that said, I appreciate your point, and thanks for the review @sharwell.